### PR TITLE
#14 - Add source priority list in AvatarConfig.

### PR DESCRIPTION
**Pull request by castriganoj**
Multiple assumptions were made in how this was implemented. The desired assumed behavior is for a user of the avatar module to pass a _sourcePriorityProperty_ as a configuration property which is used to reorder the sources constant in the avatar service. The reordering should ignore invalid or blank sources. It should only order the source values passed to it - the ordering for the remaining values should keep their default order. 

Please take a look and let me know how to proceed if this is not the desired behavior. We can also explore isolating the configuration responsibilities and or sorting responsibilities into separate services. I began writing unit tests for sorting but saw they were started here #38 for the avatar service and wanted to confirm requirements anyway. 

Thanks for the cool Angular component!

#14

### CODE REVIEW

**Comment by HaithemMosbahi** [comment]
Hi @castriganoj, so I just merged the refactoring / unit testing of the avatar service into master. Could you please take a look and adjust your pull request.
My idea is to do the overriding stuff ( avatar colors and avatar sources ) at avatar module's loading time, in other word as soon we receive the config object : 
 https://github.com/HaithemMosbahi/ngx-avatar/blob/f82321b7d3df44b333816de57fbfa109ef2a5dc1/projects/ngx-avatar/src/lib/avatar.service.ts#L54

Also, now there's an enum for avatar sources, we could also export this enum for users so it would be easier and cleaner to pass the array of sources

**Comment by castriganoj** [comment]
@HaithemMosbahi I should be able to get to this by the end of the week. Next week at the latest.

**Comment by HaithemMosbahi** [comment]
Since there hasn't been any updates on this pull request I'm gonna close it and create a new branch that will implement this feature.
I tried to take over this pr and do the necessary changes but I couldn't rebase the branch to the latest state ( I don't the right to sync the forked repo with ngx-avatar repo )

**Comment by castriganoj** [comment]
Sorry for the delay. Is this still active? I didn't see it in the base branch but if you are working on it somewhere else feel free to close the pr or let me know and I will close it and delete the branch. 

I merged upstream and removed and re-implement the initial changes in a new commit. Typescript enums can do some interesting things depending on how they are defined - https://www.typescriptlang.org/docs/handbook/enums.html. In this case the avatarSources compiles down to a javascript array and so the same sorting method to sort before can be used here. This also allows users to pass an array of sources as a string or as an enum to the config. Thanks!!

**Comment by HaithemMosbahi** [comment]
@castriganoj Thank you for your updates. I added some reviews

**Comment by castriganoj** [comment]
Unit tests have been implemented. Please review and provide feedback where needed. Best, 
Jim

File: projects/ngx-avatar/src/lib/avatar.service.spec.ts [hunk 1]
```diff
@@ -7,127 +7,264 @@ import {
...
+      httpMock.verify();
     });
 
-    it('should override the priority order when the user provide valid list of sources', () => {
-      
+    it("should be created", () => {
+      expect(avatarService).toBeTruthy();
     });
 
-    it('should ignore unknown sources', () => {
```
[the remark this answers is not in the document]
Comment (castriganoj) [reply projects/ngx-avatar/src/lib/avatar.service.spec.ts]: It was moved down to line 184. Service tests were separated into 2 sections - one for custom configuration and one for default configuration.

**Comment by HaithemMosbahi** [comment]
Already merged in #55 after some refactoring

**Pull request closed without merging**